src/main/java/com/mewna/catnip/Catnip.java

```diff
@@ -116,6 +123,67 @@ static String getShardCountUrl() {
     @Nullable
     User selfUser();
     
+    default void setPresence(@Nonnull final Presence presence) {
+        setPresence(presence, true);
+    }
+    
+    default void setPresence(@Nonnull final Presence presence, final boolean broadcastToAllShards) {
+        if (!broadcastToAllShards) {
+            final int apparentShards = shardManager().shardCount() == 0
+                    ? 0
+                    : ThreadLocalRandom.current().nextInt(shardManager().shardCount());
+            eventBus().publish(String.format("catnip:gateway:ws-outgoing:%s:presence-update", apparentShards), presence);
```

Comment:
Should use `CatnipShard#websocketMessageQueueAddress` 

https://github.com/mewna/catnip/blob/80cd084dbde7546d17565e79bab7f6f3fe2c818d/src/main/java/com/mewna/catnip/shard/CatnipShard.java#L97-L99